portfolio/src/main/java/com/google/sps/servlets/MarkerAreaServlet.java

```diff
@@ -0,0 +1,88 @@
+@WebServlet("/markersByArea")
+public class MarkerAreaServlet extends HttpServlet {
+    private static final Gson gson = new Gson();
+    private static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    private static final String ENTITY_TITLE = "Marker";
+    private static final String ENTITY_PROPERTY_KEY_1 = "lat";
+    private static final String ENTITY_PROPERTY_KEY_2 = "lng";
+    private static final String ENTITY_PROPERTY_KEY_3 = "crimeType"; 
+    private static final String ENTITY_PROPERTY_KEY_4 = "date"; 
+    private static final String ENTITY_PROPERTY_KEY_5 = "time"; 
+    private static final String ENTITY_PROPERTY_KEY_6 = "address";
+    private static final String ENTITY_PROPERTY_KEY_7 = "description";  
+
+    /** Responds with JSON array containing marker data. */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{ //add parameter(userLocation)
+        List<Marker> markers = getMarkersByArea(request);
+        String json = gson.toJson(markers);
+        response.setContentType("application/json");
+        response.getWriter().println(json); 
+    }
+
+    /** Fetches markers from Datastore. */
+    private List<Marker> getMarkersByArea(HttpServletRequest request){
+        List<Marker> markers = new ArrayList<>();
+        Query query = new Query(ENTITY_TITLE);
+        PreparedQuery results = datastore.prepare(query);
+        Location location = getUserLocation(request);
+
+        for(Entity entity: results.asIterable()){
+            double lat = (double) entity.getProperty("lat");
+            double lng = (double) entity.getProperty("lng");
+            String crime = (String) entity.getProperty("crimeType");
+            String date = (String) entity.getProperty("date");
+            String time = (String) entity.getProperty("time");
+            String address = (String) entity.getProperty("address");
+            String description = (String) entity.getProperty("description");
+            //fetches markers olny if they are inside the wanted area
+            if(inArea(location, lat, lng)){ 
+                Marker marker = new Marker(lat, lng, crime, date, time, address, description);
+                markers.add(marker);
+            }
+        }
+        return markers;
+    }
+
+    public Location getUserLocation(HttpServletRequest request){
+        String locationStr = request.getParameter("location");
+        String[] locationArrStr = locationStr.split(",", 2);
+        double lat = Double.parseDouble(locationArrStr[0]);
+        double lng = Double.parseDouble(locationArrStr[1]);
+        Location userLocation = new Location(lat, lng);
+        return userLocation;
+
+    }
+
+    public boolean inArea(Location location, double markerLat, double markerLng){
+        return haversineDistance(location, markerLat, markerLng) <= 1609.34; //return markers that are within 1 mile (1609 meters) away from user
```

Comment:
Don't use magic numbers. Create a variable for 1609.34, so readers know what this is. Then you don't even need the comment

  private static final double METERS_IN_A_MILE = 1609.34;